llvm/include/llvm/CodeGen/MachinePipeliner.h

```diff
@@ -305,6 +309,9 @@ class SwingSchedulerDAG : public ScheduleDAGInstrs {
                          SetVector<SUnit *> &NodesAdded);
   void computeNodeOrder(NodeSetType &NodeSets);
   void checkValidNodeOrder(const NodeSetType &Circuits) const;
+  bool tryScheduleWithII(SMSchedule &Schedule, unsigned II,
```

Comment:
This here is mostly refactoring IIRC. The `MachinePipeliner` is very much not modular, and this function takes a bit chunk of code out of the main loop.

The other reason is that it makes it easy to try multiple node placement strategies for the same II. Instead of trying one, and if it fails, the same placement is tried with `II+1`.